public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -285,7 +287,11 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
                   )
               );
             } else {
-              return (linkedDatasource.query(linkedRequest) as Observable<DataQueryResponse>).pipe(
+              const response = linkedDatasource.query(linkedRequest);
+              if (!('pipe' in response) || typeof response.pipe !== 'function') {
```

Comment:
Sorry one more comment here, I thought I wrote in the first batch. The response is either a Promise or Observable which is why we need to cast to use Obervable API. Both are objects representing async operation so we can also just transform Promise to Observable (as Promise is a subgroup of Observable API). So you should be able to use https://rxjs.dev/api/index/function/from which takes both Promise or Observable and returns an Observable.